### PR TITLE
fix: reward pool Calculates Incorrect Balances

### PR DESCRIPTION
**Pull request by pandadefi**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Checklist

- [x] I have run vyper and solidity linting
- [x] I have run the tests on my machine
- [x] I have followed commitlint guidelines
- [ ] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

File: contracts/RewardPool.vy [hunk 1]
```diff
@@ -222,50 +207,18 @@ def _claim(addr: address, last_token_time: uint256) -> uint256:
...
-
-        else:
-            # Calc
-            # + i * 2 is for rounding errors
-            dt: int128 = convert(week_cursor - old_user_point.ts, int128)
-            zero: int128 = 0
-            balance_of: uint256 = convert(max(old_user_point.bias - dt * old_user_point.slope, zero), uint256)
-            if balance_of == 0 and user_epoch > max_user_epoch:
+        balance_of: uint256 = VEYFI.balanceOf(addr, week_cursor)
+        if balance_of == 0 and user_epoch > max_user_epoch:
```
Comment (mariuspod) [inline contracts/RewardPool.vy:218]: observations:
- init of `user_epoch` has been removed in this commit so this check seems redundant.
- at the end of `#_claim` the user_epoch is set to `-1` and pushed into the `Claimed` event.

File: contracts/RewardPool.vy [hunk 2]
```diff
@@ -140,39 +140,6 @@ def _find_timestamp_epoch(_timestamp: uint256) -> uint256:
...
-        if pt.ts <= _timestamp:
-            _min = _mid
-        else:
-            _max = _mid - 1
-    return _min
-
-
-@view
-@external
-def ve_for_at(_user: address, _timestamp: uint256) -> uint256:
```
Comment (storming0x) [inline contracts/RewardPool.vy:162]: with the new method, is there a replacement way to get this info? seems useful for debugging issues and data off chain

**Comment by storming0x** [comment]
reran CI tests since i was getting errors on my local, and see two tests failing now

File: contracts/RewardPool.vy [hunk 3]
```diff
@@ -210,8 +177,6 @@ def checkpoint_total_supply():
 
 @internal
 def _claim(addr: address, last_token_time: uint256) -> uint256:
-    # Minimal user_epoch is 0 (if user had no point)
-    user_epoch: uint256 = 0
```
[the remark this answers is not in the document]
Comment (pandadefi) [reply contracts/RewardPool.vy:214]: Now we have the changes that are occurring in `replay_slope_changes` that are not possible here.

**Comment by pandadefi** [comment]
> reran CI tests since i was getting errors on my local, and see two tests failing now

This requires the changes from the other PRs to get green

File: contracts/RewardPool.vy (hunk 2 above)
Comment (storming0x) [reply contracts/RewardPool.vy:162]: ok just to be clear this method is now replaced with balanceOF?

Comment (pandadefi) [reply contracts/RewardPool.vy:162]: Yes

**Review by storming0x — APPROVED** [review]
LGTM!

**Pull request merged**